application/src/main/java/org/togetherjava/tjbot/commands/github/GitHubReference.java

```diff
@@ -0,0 +1,209 @@
+     * {@value Message#MAX_EMBED_COUNT} embeds at a time (the discord limit)
+     */
+    private void replyBatchEmbeds(List<MessageEmbed> embeds, Message message,
+            boolean mentionRepliedUser) {
+        List<List<MessageEmbed>> partition = ListUtils.partition(embeds, Message.MAX_EMBED_COUNT);
+        boolean isFirstBatch = true;
+
+        MessageChannel sourceChannel = message.getChannelType() == ChannelType.GUILD_PUBLIC_THREAD
+                ? message.getChannel().asThreadChannel()
+                : message.getChannel().asTextChannel();
+
+        for (List<MessageEmbed> messageEmbeds : partition) {
+            if (isFirstBatch) {
+                message.replyEmbeds(messageEmbeds).mentionRepliedUser(mentionRepliedUser).queue();
+
+                isFirstBatch = false;
+            } else {
+                sourceChannel.sendMessageEmbeds(messageEmbeds).queue();
+            }
+        }
+    }
+
+    /**
+     * Generates the embed to reply with when someone references an issue.
+     */
+    MessageEmbed generateReply(GHIssue issue) throws UncheckedIOException {
+        try {
+            String title = "[#%d] %s".formatted(issue.getNumber(), issue.getTitle());
+            String titleUrl = issue.getHtmlUrl().toString();
+            String description = issue.getBody();
+
+            String labels = issue.getLabels()
+                .stream()
+                .map(GHLabel::getName)
+                .collect(Collectors.joining(", "));
+
+            String assignees = issue.getAssignees()
+                .stream()
+                .map(this::getUserNameOrThrow)
+                .collect(Collectors.joining(", "));
+
+            Instant createdAt = issue.getCreatedAt().toInstant();
+            String dateOfCreation = FORMATTER.format(createdAt);
+
+            String footer = "%s • %s • %s".formatted(labels, assignees, dateOfCreation);
+
+            return new EmbedBuilder()
+                .setColor(issue.getState() == GHIssueState.OPEN ? ISSUE_OPEN : ISSUE_CLOSE)
+                .setTitle(title, titleUrl)
+                .setDescription(description)
+                .setAuthor(issue.getUser().getName(), null, issue.getUser().getAvatarUrl())
+                .setFooter(footer)
+                .build();
+
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex);
+        }
+    }
+
+    /**
+     * Either properly gathers the name of a user or throws a UncheckedIOException.
+     */
+    private String getUserNameOrThrow(GHUser user) throws UncheckedIOException {
+        try {
+            return user.getName();
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex);
+        }
+    }
+
+    /**
+     * Looks through all of the given repositories for an issue/pr with the given id.
+     */
+    Optional<GHIssue> findIssue(int id) {
+        return repositories.stream().map(repository -> {
+            try {
```

Comment:
changing to sequential stream might have caused some problems, searches are inaccurate for some reasons. Switching back to parallel stream resolves the issue for `/github-search` but it shows inconsistent behaviour for `#issueNumber` based searches since we have multiple repsitories, so sometimes it will link issue number 11 from tjbot repo, sometimes from tjplays repo if there's a match. I'm trying to find a way to make it consistent.

PR is not up for merge yet.

Reply:
> changing to sequential stream might have caused some problems, searches are inaccurate for some reasons. Switching back to parallel stream resolves the issue for `/github-search`

This sounds very wrong to me, the error must lie somewhere else

> but it shows inconsistent behaviour for #issueNumber based searches since we have multiple repsitories, so sometimes it will link issue number 11 from tjbot repo, sometimes from tjplays repo if there's a match. I'm trying to find a way to make it consistent. Although im not sure if this is a concern, since issue numbers won't usually match for current number of issues across repos.
TJ-Bot is way ahead in issue numbers than modernjava or tjplays.

This probably comes from using `findAny`, you might want to use the config for the order, which to prefer.

Reply:
i'm investing the underlying problem, but this was just an observation of mine due to a recent change.

Reply:
found out the cause, since both features were finding issues by id so it was defaulting to first repo aka tj-bot. Now github search also uses title along with id to match whereas github reference uses default repo id to search results accordingly. New changes resolves both issues.